### PR TITLE
FM2-534: Searching should provide method to ensure consistent sort order

### PR DESCRIPTION
**Pull request by mogoodrich**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/module/fhir2/api/search/EncounterSearchQueryTest.java [hunk 1]
```diff
@@ -1539,4 +1539,21 @@ public void shouldReturnCollectionOfEncountersSortedByEncounterDate() {
 			    greaterThanOrEqualTo(resultList.get(i).getPeriod().getStart()));
 		}
 	}
+	
+	@Test
+	public void searchForEncounters_shouldReturnAllNonVoidedEncountersSortedById() {
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/fhir2/api/search/EncounterSearchQueryTest.java:1544]: FWIW, this test was passing prior to this commit, but it does codify the behavior we now expect to see, that it sorts by id.

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPersonDaoImpl.java [hunk 2]
```diff
@@ -88,5 +88,5 @@ protected String getSqlAlias() {
 	protected void handleVoidable(Criteria criteria) {
 		criteria.add(eq("personVoided", false));
 	}
-
+
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPersonDaoImpl.java:91]: Just some autoformatting I must have missed in my last commit.

**Review by ibacher — APPROVED** [review]

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseFhirDao.java [hunk 3]
```diff
@@ -163,6 +163,7 @@ public List<Integer> getSearchResultIds(@Nonnull SearchParameterMap theParams) {
 		setupSearchParams(criteria, theParams);
 		handleSort(criteria, theParams.getSortSpec());
 		
+		criteria.addOrder(org.hibernate.criterion.Order.asc("id"));
```
[the remark this answers is not in the document]
Comment (mogoodrich) [reply api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseFhirDao.java:166]: True, but my thought was to make this as close as possible to what I'd expect MySQL to do by default, which would be to return in ascending order via primary key, all things being equal.  Any thoughts @mseaton ?

**Review by mseaton — APPROVED** [review]

**Comment by mogoodrich** [comment]
Thanks @ibacher @mseaton , I'm going to merge this in as-is...

**Pull request merged**